### PR TITLE
Implement retries for Start Cloud Slack func

### PR DESCRIPTION
**Pull request by mszostok**
## Description

Changes proposed in this pull request:

- Implement retries for Start Cloud Slack func

## Testing

- Connect Instance and restart Slack Router

### CODE REVIEW

File: cmd/botkube/main.go [hunk 1]
```diff
@@ -125,9 +125,9 @@ func run(ctx context.Context) error {
 	errGroup, ctx := errgroup.WithContext(ctx)
 	defer func() {
 		err := errGroup.Wait()
-		if err != nil {
-			wrappedErr := reportFatalError("while waiting for goroutines to finish gracefully", err)
-			logger.Errorf(wrappedErr.Error())
+		wrappedErr := reportFatalError("while waiting for goroutines to finish gracefully", err)
+		if wrappedErr != nil {
```
Comment (mszostok) [inline cmd/botkube/main.go:129]: I got panic here as the wrappedErr can e.g. return nil for context cancel related errors, so I change the logic here.

File: pkg/bot/cloudslack.go [hunk 2]
```diff
@@ -32,8 +34,11 @@ import (
 )
 
 const (
-	APIKeyContextKey       = "X-Api-Key"       // #nosec
-	DeploymentIDContextKey = "X-Deployment-Id" // #nosec
+	APIKeyContextKey        = "X-Api-Key"       // #nosec
+	DeploymentIDContextKey  = "X-Deployment-Id" // #nosec
+	retryDelay              = time.Second
+	maxRetries              = 25
+	successIntervalDuration = 5 * time.Minute
```
Comment (pkosiec) [inline pkg/bot/cloudslack.go:41]: WDYT about 1 or 2 minutes?

**Review by pkosiec — APPROVED** [review]
Works flawlessly 👌

File: pkg/bot/cloudslack.go (hunk 2 above)
Comment (mszostok) [reply pkg/bot/cloudslack.go:41]: discussed f2f, randomly picked 3 😄  for now we don't have enough data to come up with well explained defaults

**Pull request merged**